### PR TITLE
Deploy fuel-core-dev1 When Merging To Master

### PR DESCRIPTION
**Pull request by ellioty**

### CODE REVIEW

**Comment by rfuelsh** [comment]
@ellioty - is there a github workflow that run to showcase how this action works?

**Comment by Voxelot** [comment]
blocked until we have repo secret key added to fuel-core

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -358,7 +372,7 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: type=gha
           cache-to: type=gha,mode=max
-      
+
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:375]: hmmm

**Comment by adlerjohn** [comment]
I've added the token

**Comment by Voxelot** [comment]
I'm thinking this should be a separate job/stage than "build-publish-master-image" since they're fairly distinct tasks. We could also consider making the "build-publish-master-image" job run on master by default, but on other branches via workflow dispatch. This would allow for deployments from feature branches without having to merge to master first.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -254,16 +260,36 @@ jobs:
...
-          notify_when: 'failure'
+          notification_title: "{workflow} has {status_message}"
+          message_format: "{emoji} *{workflow}* {status_message} in <{repo_url}|{repo}> : <{run_url}|View Run Results>"
+          footer: ""
+          notify_when: "failure"
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.ACTION_MONITORING_SLACK }}
+
+  deploy:
+    # Deploy Latest Fuel-Core Release
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:278]: Should this have a `needs`?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -198,21 +204,21 @@ jobs:
...
         uses: ravsamhq/notify-slack-action@v1
         if: always()
         with:
           status: ${{ job.status }}
           token: ${{ secrets.GITHUB_TOKEN }}
-          notification_title: '{workflow} has {status_message}'
-          message_format: '{emoji} *{workflow}* {status_message} in <{repo_url}|{repo}> : <{run_url}|View Run Results>'
-          footer: ''
-          notify_when: 'failure'
+          notification_title: "{workflow} has {status_message}"
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:215]: As tempting as it is to drive-by fix formatting, do this in a separate PR.

Comment (ellioty) [reply .github/workflows/ci.yml:215]: Seems like VS Code formatted it based on YAML rules, is it okay to leave or would you like me to still revert?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Voxelot) [reply .github/workflows/ci.yml:278]: I think this should be automatic on every merge to master, embracing the "continuous deployment" mindset.

Comment (ellioty) [reply .github/workflows/ci.yml:278]: Sounds good, let's leave this as is and evaluate this again if needed

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -254,16 +260,36 @@ jobs:
...
           SLACK_WEBHOOK_URL: ${{ secrets.ACTION_MONITORING_SLACK }}
+
+  deploy:
+    # Deploy Latest Fuel-Core Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Deploy Fuel-Core Developer Environment
+        uses: peter-evans/repository-dispatch@v2
+        if: github.event.inputs.deploy-dev-env == 'true'
```
Comment (Voxelot) [inline .github/workflows/ci.yml:284]:
```suggestion
        if: github.ref == 'refs/heads/master'
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -1,6 +1,12 @@
 name: CI
 
 on:
+  workflow_dispatch:
+    inputs:
+      deploy-dev-env:
+        description: "Flag to deploy to Dev environment"
+        type: boolean
+        required: false
```
Comment (Voxelot) [inline .github/workflows/ci.yml:9]:
```suggestion
```

File: .github/workflows/ci.yml (hunk 3 above)
Comment (adlerjohn) [reply .github/workflows/ci.yml:210]: Rather than reverting, make a new PR against `master` that performs the formatting, merge that in, then merge `master` into this branch. Boom, now all your formatting changes just get cancelled out for free!

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -254,16 +255,36 @@ jobs:
...
+        if: github.ref == 'refs/heads/master'
+        with:
+          # Personal Access Token to trigger repository_dispatch actions
+          token: ${{ secrets.REPO_TOKEN }}
+          repository: FuelLabs/fuel-deployment
+          event-type: deploy-fuel-core-dev
+          client-payload: '{ "k8s-type": "${{ env.K8S }}", "config-directory": "${{ env.CONFIG }}", "config-env": "${{ env.ENV }}", "delete-infra": "${{ env.DELETE_INFRA }}" }'
         env:
-          SLACK_WEBHOOK_URL: ${{ secrets.ACTION_MONITORING_SLACK }}
+          CONFIG: "fuel-dev1"
```
Comment (rfuelsh) [inline .github/workflows/ci.yml:287]: @ellioty - do these values stay static? I guess it just for developer purposes

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ellioty) [reply .github/workflows/ci.yml:210]: https://github.com/FuelLabs/fuel-core/pull/360

Comment (ellioty) [reply .github/workflows/ci.yml:210]: https://github.com/FuelLabs/fuel-core/pull/360

**Comment by rfuelsh** [comment]
@ellioty - the remote PR workflow works and I confirmed this- https://github.com/FuelLabs/fuel-core/runs/6796584537?check_suite_focus=true 

please re-run this and lets see why issues come up

**Review by rfuelsh — APPROVED** [review]
